### DISCUSSION TITLE
Check if user is logged in

### DISCUSSION THREAD

**Discussion by MatejaSrejic** [kreait/firebase-php#529] [Q&A]
Noob here, I'm wondering how can we check if a user is logged in?
I found this section on the docs, but it seems to throw an error.

![image](https://user-images.githubusercontent.com/75619038/102600813-9dabb680-411f-11eb-8248-b482ee3fb625.png)

The error I'm getting is `Parse error: syntax error, unexpected ')' in D:\PHP\auth\__\login\index.php on line 8`
The code:
```
<?php
use Kreait\Firebase\Factory;
use Kreait\Firebase\Auth;

require_once('vendor/autoload.php');
$factory = (new Factory)->withServiceAccount('myCredentials.json');
$auth = $factory->createAuth();
$signInResult = $auth->signIn*(); // this is line 8

echo $signInResult->idToken();
```

**Comment by jeromegamez** [comment]
`$auth->signIn*` was just a shorthand for the methods described in https://firebase-php.readthedocs.io/en/5.14.0/authentication.html#custom-authentication-flows , so using this explicitly will not work, but e.g. `$auth->signInAnonymously()` will

<img width="744" alt="grafik" src="https://user-images.githubusercontent.com/67554/102601799-f3cd2980-4120-11eb-83bf-1a6095e64162.png">

Please note that using one of these methods will sign in the user, so you can't check if a user is signed in with this (because you just signed them in 😅 )

**Reply by MatejaSrejic** [reply]
Oh okay thanks for the explanation.
Warned you I'm a newbie 😅

So is there any other way then to check if the user is signed in or not?

**Reply by jeromegamez** [reply]
Given you have a user ID, you can use it to get a user record, which contains a `lastLoginAt` field.

```php
$uid = '...';

$userRecord = $auth->getUser($uid);

$lastLoginAt = $user->metadata->lastLoginAt;
```
